runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -301,14 +303,21 @@ public void committed(long epoch, Layout layout, boolean force)
                 })
                 .toArray(CompletableFuture[]::new);
 
+        // Make sure a quorum gets to commit.
         int timeouts = 0;
-        int responses = 0;
-        while (responses < commitList.length) {
-            // wait for someone to complete.
+        long wrongEpochRejected = 0L;
+        while (true) {
+            if (commitList.length < getQuorumNumber()) {
+                log.debug("commit: Quorum unreachable, remaining={}, required={}", commitList,
```

Comment:
commitList is based on a layout passed to the method, which can be different from the layout used by getQuorumNumber

Reply:
Yep this is wrong and should be fixed.